lib/testers/rule-tester.js

```diff
@@ -446,7 +446,7 @@ RuleTester.prototype = {
                         }
 
                         if (item.errors[i].type) {
-                            assert.equal(messages[i].nodeType, item.errors[i].type, `Error type should be ${item.errors[i].type}`);
```

Comment:
I'm not sure that removing this message is the right way to go about this change; I think "Error type should be CallExpression" provides useful context about the issue, whereas "CallExpression != SomethingElse" is sort of confusing.

Instead, maybe the message should say something along the lines of: "Expected an error type of CallExpression but found Indentifier". That keeps the helpful "Error type should be" context, while also providing the additional information in https://github.com/eslint/eslint/issues/6106.